Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorHelpers.cpp

```diff
@@ -382,11 +382,28 @@ namespace AzToolsFramework
 
     bool EditorHelpers::IsSelectableInViewport(const AZ::EntityId entityId) const
     {
-        return IsSelectableAccordingToFocusMode(entityId) && IsSelectableAccordingToContainerEntities(entityId);
+        auto entityCacheId = m_entityDataCache->GetVisibleEntityIndexFromId(entityId);
+        if (entityCacheId.has_value())
```

Comment:
nit: Also I can't help mentioning it but these could all be if initializers 🙂 

```
if (auto entityCacheId = m_entityDataCache->GetVisibleEntityIndexFromId(entityId);
    entityCacheId.has_value())
```

Reply:
Done!